core/src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapperLegacy.java

```diff
@@ -0,0 +1,447 @@
+                                               DoubleFieldMapper lonMapper, StringFieldMapper geohashMapper, MultiFields multiFields, Explicit<Boolean> ignoreMalformed,
+                                               CopyTo copyTo) {
+            fieldType.setTokenized(false);
+            setupFieldType(context);
+            fieldType.setHasDocValues(false);
+            defaultFieldType.setHasDocValues(false);
+            return new GeoPointFieldMapperLegacy(simpleName, fieldType, defaultFieldType, indexSettings, pathType, latMapper, lonMapper,
+                    geohashMapper, multiFields, ignoreMalformed, coerce(context), copyTo);
+        }
+
+        @Override
+        public GeoPointFieldMapperLegacy build(BuilderContext context) {
+            return super.build(context);
+        }
+    }
+
+    public static Builder parse(Builder builder, Map<String, Object> node, Mapper.TypeParser.ParserContext parserContext) throws MapperParsingException {
+        final boolean indexCreatedBeforeV2_0 = parserContext.indexVersionCreated().before(Version.V_2_0_0);
+        for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+            Map.Entry<String, Object> entry = iterator.next();
+            String propName = Strings.toUnderscoreCase(entry.getKey());
+            Object propNode = entry.getValue();
+            if (indexCreatedBeforeV2_0 && propName.equals("validate")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lon")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("validate_lat")) {
+                builder.ignoreMalformed = !XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (propName.equals(Names.COERCE)) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lat")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            } else if (indexCreatedBeforeV2_0 && propName.equals("normalize_lon")) {
+                builder.coerce = XContentMapValues.nodeBooleanValue(propNode);
+                iterator.remove();
+            }
+        }
+        return builder;
+    }
+
+    public static final class GeoPointFieldType extends BaseGeoPointFieldType {
```

Comment:
This looks like the exact same implementation...can `BaseGeoPointFieldType` just be renamed and used in both classes? There isn't a reason the field type needs to be a static inner class of the mapper, they are just there for convenience since they are so small and tightly coupled with the mapper parsing code.